### PR TITLE
Add dev docs for entry point

### PR DESCRIPTION
**Pull request by ntouran**
Also add intersphinx plugin to get good link to Python docs.

### CODE REVIEW

File: doc/developer/entrypoints.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    * ``"required"`` if a settings input file must be provided,
+    * ``"optional"`` if it may be provided but not required, 
+    * ``None`` if no settings input is allowed
+
+
+.. tip:: ARMI apps often collect EntryPoints in a ``cli/`` directory (Command Line Interface)
+
+Next, implement the :py:meth:`~armi.cli.entryPoint.EntryPoint.addOptions` method. Here you can both:
+
+* turn various App Settings into command-line arguments with :py:meth:`~armi.cli.entryPoint.EntryPoint.createOptionFromSetting`
```
Comment (youngmit) [inline doc/developer/entrypoints.rst:29]: ~~App~~ Settings

File: doc/developer/entrypoints.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+* add arbitrary other command-line arguments using the standard :py:mod:`python:argparse` library.
```
Comment (youngmit) [inline doc/developer/entrypoints.rst:30]: arbitrary ~~other~~ command-line

File: doc/developer/entrypoints.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+The values of the non-setting arguments will become attributes in ``self.args`` for later use.
```
Comment (youngmit) [inline doc/developer/entrypoints.rst:32]: Now that intersphinx is set up, we might want to link the relevant `argparse` docs

File: doc/developer/entrypoints.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
+Entry Points
+============
+
+**Entry Points** are extensible routines that you can choose
+when you invoke your application. The :py:mod:`built-in entry points <armi.cli>`
```
Comment (youngmit) [inline doc/developer/entrypoints.rst:5]: Maybe something to the tune of "entry points are like the verbs that your App can *do*," and "are invoked by running `> myapp [verb]`"

File: doc/developer/entrypoints.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        Long description of entry point here. This will actually get picked up and used
+        as the help text on the command line itself!
+        """
+
+        name = "do-my-thing"
+        settingsArgument = "required"
+
+        def addOptions(self):
+            self.createOptionFromSetting(CONF_DRAGON_PATH)
+            self.createOptionFromSetting(CONF_DRAGON_DATA_PATH)
```
Comment (youngmit) [inline doc/developer/entrypoints.rst:56]: These settings dont actually exist in the framework; might be better to make an example with framework settings

**Review by youngmit — CHANGES_REQUESTED** [review]
Mostly minor copy edits. Should use framework settings in the example.

File: armi/cli/run.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Run ARMI."""
+"""Run full ARMI case."""
```
[the remark this answers is not in the document]
Comment (ntouran) [reply armi/cli/run.py:15]: ok

File: doc/developer/entrypoints.rst (hunk 3 above)
Comment (ntouran) [reply doc/developer/entrypoints.rst:32]: that's what this does, and that's why I added intersphinx.

Comment (youngmit) [reply doc/developer/entrypoints.rst:32]: oh im a dummy, its right there. i didnt see the link 🤦‍♂️

**Review by youngmit — APPROVED** [review]

**Pull request merged**